lib/dry/schema/rule_applier.rb

```diff
@@ -32,7 +32,7 @@ def call(input)
         rules.each do |name, rule|
           next if input.error?(name)
 
-          result = rule.(input.to_h)
+          result = rule.(deep_symbolize_hash(input.to_h))
```

Comment:
keys are meant to be coerced by the key map object, this is a separate step so here we expect the hash to already have symbol keys